### ISSUE TITLE
Classes with URL in them will give warning  illegal reflective

### ISSUE THREAD

**Issue by Arbenit** [jqno/equalsverifier#340]
# What steps will reproduce the problem?
Create class with URL 

# What is the code that triggers this problem?
```
    @Test
    public void testEqualsAndHashCode(){
        EqualsVerifier.forClass(Pojo.class).usingGetClass().verify();
    }
```

Provide an example of a complete class (equals method, hashCode method, relevant fields) and a call to EqualsVerifier.
```
import javax.validation.constraints.NotNull;
import java.net.URL;
import java.util.Objects;

public class Pojo {

    private final URL url;

    public Pojo(@NotNull final URL url) {
        this.url = url;
    }

    @NotNull
    public URL getUrl() {
        return url;
    }


    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Pojo pojo = (Pojo) o;
        return Objects.equals(url, pojo.url);
    }

    @Override
    public int hashCode() {
        return Objects.hash(url);
    }


}
````
```
public class PojoTest {

    static {
        try {
            URL = new URL("https://example.com/");
        } catch (MalformedURLException e) {
            throw new IllegalStateException(e);
        }
    }

    private final static URL URL;

    Pojo pojo;

    @BeforeEach
    void setUp() throws MalformedURLException {
        pojo = new Pojo(URL);
    }

    @Test
    public void testEqualsAndHashCode(){
        EqualsVerifier.forClass(Pojo.class).usingGetClass().verify();
    }
}
```

Remember, you have the problem right in front of you. I have to try and reproduce it in my spare time. Please help me, so that I can help you!

# What error message or stack trace does EqualsVerifier give?

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (file:/userspace/nl/jqno/equalsverifier/equalsverifier/3.4.2/equalsverifier-3.4.2.jar) to field java.net.UrlDeserializedState.protocol
WARNING: Please consider reporting this to the maintainers of nl.jqno.equalsverifier.internal.reflection.FieldAccessor
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

# What did you expect?

no warning

# Which version of EqualsVerifier are you using?

3.4.2

# Please provide any additional information below.

```    
@Test
    public void testEqualsAndHashCode() throws MalformedURLException {
        EqualsVerifier.forClass(Pojo.class).usingGetClass()
                      .withPrefabValues(URL.class,new URL("http://www.example.com/"), new URL("http://www.google.ca/")).verify();
    }
```
This will change the warning to 

`
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (file:/userspace/nl/jqno/equalsverifier/equalsverifier/3.4.2/equalsverifier-3.4.2.jar) to field java.net.URL.protocol
WARNING: Please consider reporting this to the maintainers of nl.jqno.equalsverifier.internal.reflection.FieldAccessor
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
`

**Comment by jqno** [comment]
I've added prefab values for `java.net.URL`. I have some other things I want to address, I'll let you know when I make a release.

**Comment by Arbenit** [comment]
Sorry for the late response but I think you might have to do more then just add a prefab value for java.net.url if you look at the additional information part when I added the withPrefabValues for the url the warning changed.

**Comment by jqno** [comment]
Can you confirm that please? I've got the EqualsVerifier build running with `--illegal-access=deny` on a bunch of different JVMs, and it has not complained.

**Comment by Arbenit** [comment]
![warning](https://user-images.githubusercontent.com/33850571/91723096-3fa39500-eb69-11ea-8a88-c433eb63b020.png)
THe warning only happens when I manually run the test ie click the run test button on either that test or the whole class. If I run it with maven verify instead it doesnt generate any warnings.

**Comment by jqno** [comment]
Ah, that at the same time explains and doesn't explain things...
It explains why I didn't see the second warning. But why would running the test from IntelliJ give such different behaviour? That said, I need to be able to reproduce this from Maven as well if I'm going to write a test and a fix for it...

**Comment by Arbenit** [comment]
I dont really know how to get the warning to show up for the maven verify so Ill close the ticket and if in the future I run into the issue again but it also shows up on maven then Ill reopen the issue with the steps required.

**Comment by jqno** [comment]
I've just released version 3.4.3, which contains the prefab value.
